src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/status/StatusBarWidget.java

```diff
@@ -105,15 +114,51 @@ else if (type == StatusBar.SCOPE_SLIDE)
       else
          scopeIcon_.setResource(CodeIcons.INSTANCE.function());
    }
+   
+   public void showMessage(String message)
+   {
+      scope_.setVisible(false);
+      panel_.setCellWidth(scope_, "0");
+      
+      message_.setVisible(true);
+      panel_.setCellWidth(message_, "100%");
+      
+      message_.setValue(message);
+      if (handler_ != null)
+      {
+         handler_.removeHandler();
+         handler_ = null;
+      }
+      
+      handler_ = Event.addNativePreviewHandler(new NativePreviewHandler()
+      {
+         
+         @Override
+         public void onPreviewNativeEvent(NativePreviewEvent event)
```

Comment:
For the rename status bar clearing, what's the gesture to stop the rename
itself (Enter or Esc?). I think once they stop editing the variable then
you can clear it (one heuristic might be that as soon as you drop back to 1
cursor you always clear any message in the status bar)

On Tue, Jul 7, 2015 at 1:51 PM, Kevin Ushey <notifications@github.com>
wrote:

> In
> src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/status/StatusBarWidget.java
> <https://github.com/rstudio/rstudio/pull/454#discussion_r34068848>:
>
> > +
> > +      message_.setVisible(true);
> > +      panel_.setCellWidth(message_, "100%");
> > +
> > +      message_.setValue(message);
> > +      if (handler_ != null)
> > +      {
> > +         handler_.removeHandler();
> > +         handler_ = null;
> > +      }
> > +
> > +      handler_ = Event.addNativePreviewHandler(new NativePreviewHandler()
> > +      {
> > +
> > +         @Override
> > +         public void onPreviewNativeEvent(NativePreviewEvent event)
>
> Good point. There are a couple of other options here:
>
>    1. Take over Enter and have it 'end' the rename (without inserting a
>    newline),
>    2. End on mouse clicks,
>    3. End if focus lost.
>
> I'll add these, but can you think of any other candidates?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/454/files#r34068848>.
>